bftengine/src/bftengine/RequestsBatchingLogic.cpp

```diff
@@ -94,7 +120,14 @@ PrePrepareMsg *RequestsBatchingLogic::batchRequests() {
     case BATCH_BY_REQ_NUM: {
       lock_guard<mutex> lock(batchProcessingLock_);
       prePrepareMsg = replica_.buildPrePrepareMsgBatchByRequestsNum(maxNumOfRequestsInBatch_);
-      if (prePrepareMsg) timers_.reset(batchFlushTimer_, milliseconds(batchFlushPeriodMs_));
+      if (prePrepareMsg) {
+        timers_.reset(batchFlushTimer_, milliseconds(batchFlushPeriodMs_));
+        auto period =
+            std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - start_timer_)
+                .count();
```

Comment:
Where do you keep the "2-seconds" history collection period?

Reply:
Currently, it's not a strict 2 seconds, I did it 20 times flush time.